fink_science/kilonova/PredictLightCurve.py

```diff
@@ -0,0 +1,222 @@
+from LightCurve import LightCurve
+import numpy as np
+from scipy.optimize import minimize
+
+import os
+from fink_science import __file__
+
+
+def calc_prediction(coeff, PCs, bias=None):
+    """ prediction helper
+
+        Parameters
+        ----------
+        coeff: np.array
+            initial prediction
+        PCs: 1D np.array
+            Principal components generated for each band
+        bias:
+            adds bias to the predictions. zero by default
+
+        Returns
+        ----------
+        error: 1D np.array
+            curve predicted with current set of coefficients
+    """
+    predicted_lc = np.zeros_like(PCs.shape[1])
+    for a, b in zip(PCs, coeff):
+        predicted_lc = np.add(predicted_lc, b * a)
+    if bias is not None:
+        predicted_lc = predicted_lc + bias
+    return predicted_lc
+
+
+def calc_loss(coeff, PCs, light_curve_seg, bias=None):
+    """ Loss function
+
+        Parameters
+        ----------
+        coeff: np.array
+            initial prediction
+        PCs: 1D np.array
+            Principal components generated for each band
+        light_curve_seg: np.array
+            actual flux measurements binned in segments of 2 days each.
+            If no measurement is made in a bin, value should be set to 0
+        bias:
+            adds bias to the predictions. zero by default
+
+        Returns
+        ----------
+        error: np.float64
+            least square error between prediction and actual
+    """
+    index = light_curve_seg != 0
+    y_pred = calc_prediction(coeff, PCs, bias=bias)
+    diff = light_curve_seg - y_pred
+    neg_index = y_pred < 0
+    diff = diff[index | neg_index]
+    error = np.sum(np.square(diff, diff))
+    return error
+
+
+class PredictLightCurve:
+
+    def __init__(self, data, object_id, num_pc_components=3):
+
+        self.lc = LightCurve(data, object_id)
+        self.current_date = None
+        self.num_pc_components = num_pc_components
+        self.bands = None
+        self.pcs = None
+
+        self.min_flux_threshold = 0
+        self.num_prediction_points = 51
```

Comment:
Why this number? where does it come from?

Reply:
the predictions generated are for a time duration of 100 days with one data point every 2 days. 
Therefore, there is a total of 51 points (100 days) in the predicted light curve. This, of course, is the result of the length of templates (number of points) chosen to fit the curves.
I hope this answers your question.

Reply:
Thanks! Gotcha this is for the template, not the real data.

Reply:
yes! It is not related to the real data points.